README.md

```diff
@@ -140,6 +140,14 @@ request state:
   mode (see `config::staged_run` and `config::guarded_run`). The bot
   removes this label when either the PR was successfully merged or its
   staging results are no longer fresh/applicable.
+* `M-abandoned-staging-checks`: Anubis discovered a stale _staging commit_.
+  This usually happens when either the PR state changes (e.g., the author
+  commits new code) or the staging_branch is unexpectedly modified. After
```

Comment:
> there is nothing extraordinary in modifying the PR branch during staging

Agreed, but that is not the event the "unexpected" applies to. The text talks about unexpected modification of the staging branch, not the PR branch. When we talk about the staging branch, we expect that only Anubis can modify it and that once Anubis stages a PR it does not stage any other PR (without having to go through the labeling code of the staged PR), right?

Are there any _expected_ modifications of the _staging_ branch that would still result in a `M-abandoned-staging-checks` label?

Reply:
> The text talks about unexpected modification

Ah, I see. Yes, such external modifications of auto branch (performed not by the bot) will result in `M-abandoned-staging-checks`. We should leave the wording it as is then.

Reply:
Great, the ball is on your side then.

Reply:
> Are there any expected modifications of the staging branch 

If by 'expected' we mean 'performed by Anubis' - I think such modifications exist. For example, an exception/assertion in the bot leads to the PR abandoning and switching to another PR, staging it, and modifying auto branch. However, we can treat this emergency situation as 'unexpected' (though Anubis is aware of it).

Reply:
> However, we can treat this emergency situation as 'unexpected' (though Anubis is aware of it).

Yes, failed assertions can be classified as unexpected events, regardless of how they are handled. However, I am not sure they are actually relevant here because, after #36, when there is an assertion, there will be an `M-failed-staging-other` or `M-failed-other` label _instead_ of the `M-abandoned-staging-checks` label, right? In other words, the new `M-abandoned-staging-checks` label cannot appear for PR X _because_ Anubis asserted while processing PR X, correct?

Reply:
> after #36, when there is an assertion, there will be an M-failed-staging-other or M-failed-other label instead of the M-abandoned-staging-checks label,

Agreed, this is not relevant - `M-abandoned-staging-checks` will not appear in a case of assertion because the only place it is applied now - after the discovery of a stale PR. For now (i.e, without PR36 changes), assertions will result in `M-failed-other`.